main/test/ca/uwaterloo/flix/language/feature/Test.Expression.NewChannel.flix

```diff
@@ -0,0 +1,175 @@
+/////////////////////////////////////////////////////////////////////////////
+// Tests the new Channel function that creates a new Channel               //
+//                                                                         //
+/////////////////////////////////////////////////////////////////////////////
+
+/************ Channels with default buffersize ***********/
+enum Color {
+    case Red,
+    case Green,
+    case Blue
+}
+
+@test
+def testNewChannel01(): Unit =
+  let x = channel Int;
+  ()
+
+@test
+def testNewChannel02(): Channel[Int] = channel Int
+
+@test
+def testNewChannel03(): Channel[Bool] = channel Bool
+
+@test
+def testNewChannel04(): Channel[Char] = channel Char
+
+@test
+def testNewChannel05(): Channel[Int8] = channel Int8
+
+@test
+def testNewChannel06(): Channel[Int16] = channel Int16
+
+@test
+def testNewChannel07(): Channel[Int32] = channel Int32
+
+@test
+def testNewChannel08(): Channel[Int64] = channel Int64
+
+@test
+def testNewChannel09(): Channel[BigInt] = channel BigInt
+
+@test
+def testNewChannel10(): Channel[Float32] = channel Float32
+
+@test
+def testNewChannel11(): Channel[Float64] = channel Float64
+
+@test
+def testNewChannel12(): Channel[Str] = channel Str
+
+@test
+def testNewChannel13(): Channel[(Int, Int)] = channel (Int, Int)
+
+//@test
+//def testNewChannel14(): Channel[Opt[Int]] = channel Opt[Int]
+
+@test
+def testNewChannel15(): Channel[List[Int]] = channel List[Int]
+
+//@test
+//def testNewChannel16(): Channel[Vec[Int]] = channel Vec[Int]
+
+@test
+def testNewChannel17(): Channel[Set[Int]] = channel Set[Int]
+
+@test
+def testNewChannel18(): Channel[Color] = channel Color
+
+@test
+def testNewChannel19(): Channel[Int -> Int] = channel Int -> Int
+
+@test
+def testNewChannel20(): Channel[(Int, Str) -> Int] = channel (Int, Str) -> Int
+
+@test
+def testNewChannel21(): Channel[Int -> (Str -> Int)] = channel Int -> Str -> Int
+
+@test
+def testNewChannel22(): Channel[Channel[Int]] = channel Channel[Int]
+
+@test
+def testNewChannel23(): Channel[Channel[Channel[Int]]] = channel Channel[Channel[Int]]
+
+@test
+def testNewChannel24(): Channel[Channel[Channel[Channel[Int]]]] = channel Channel[Channel[Channel[Int]]]
```

Comment:
I think you could have added even more test cases in a systematic way. You should have done e.g.:

Channel[Channel[Bool]]
Channel[Channel[Int8]]
Channel[Channel[Int16]]

etc.

It is very valuable to be extremely systematic. Even if it is tedious.